### ISSUE TITLE
Helm 3: `helm install` and `helm upgrade` import values differently for aliased sub-charts

### ISSUE THREAD

**Issue by hd-rk** [helm/helm#6001]
Output of `helm version`:
```
version.BuildInfo{Version:"v3.0.0-alpha.2", GitCommit:"97e7461e41455e58d89b4d7d192fed5352001d44", GitTreeState:"clean", GoVersion:"go1.12.7"}
```

Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.0", GitCommit:"fc32d2f3698e36b93322a3465f63a14e9f0eaead", GitTreeState:"clean", BuildDate:"2018-03-27T00:13:18Z", GoVersion:"go1.9.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"13+", GitVersion:"v1.13.7-gke.8", GitCommit:"7d3d6f113e933ed1b44b78dff4baf649258415e5", GitTreeState:"clean", BuildDate:"2019-06-19T16:37:16Z", GoVersion:"go1.11.5b4", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): GKE

Sample Helm project https://github.com/hd-rk/helm-examples/tree/develop/v3-alias-values

In this setup, `parent` chart has `child` as a sub-chart under the alias `childzzz`. I observed there's a discrepancy in how `helm install` and `helm upgrade` import values to be used in the sub-chart.
```
dependencies:
- name: child
  version: 0.1.0
  repository: file://../child
  alias: childzzz
```

`helm install` uses the map under `child` (the sub-chart `name` field in the parent `Chart.yaml`) to render sub-chart templates.

`helm upgrade` uses the map under `childzzz` (the sub-chart `alias` field in the parent `Chart.yaml`) to render sub-chart templates.

**Comment by hd-rk** [comment]
This issue is likely due to the fact that `ProcessDependencies` (which calls `getAliasDependency`) is not called in `helm install`, while it is called in `helm upgrade`.

https://github.com/helm/helm/blob/ff818fceca20a4a5414fa26bcf63a305c6ef9343/pkg/chartutil/dependencies.go#L27-L32

**Comment by hickeyma** [comment]
@hd-rk can you provide more details on how this shows itself? Maybe some details on the output to get better idea of the issue?

**Comment by hd-rk** [comment]
@hickeyma I can briefly explain here, you can also take a look at the sample project and the README in it.

I have a `parent` chart that depends on a `child` chart with an `alias`
```
dependencies:
- name: child
  version: 0.1.0
  repository: file://../child
  alias: childzzz
```
`child` has a `ConfigMap` template:
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: {{ template "child.fullname" . }}-config
  labels:
{{ include "child.labels" . | indent 4 }}
data:
  foo: {{ .Values.bar | quote }}
```
I deploy `parent` using `helm install test charts/parent -f values.override.yaml` with the values file:
```
child:
  bar: something
childzzz:
  bar: something_else
```
The deployed `ConfigMap` looks like this, you can see `{{ .Values.bar | quote }}` is rendered to `something`
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: test-child-config
  labels:
    app.kubernetes.io/name: child
    helm.sh/chart: child-0.1.0
    app.kubernetes.io/instance: test
    app.kubernetes.io/version: "1.16.0"
    app.kubernetes.io/managed-by: Helm
data:
  foo: "something"
```
Then, running `helm upgrade test charts/parent -f values.override.yaml` will render a completely different `ConfigMap`
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: test-childzzz-config
  labels:
    app.kubernetes.io/name: childzzz
    helm.sh/chart: childzzz-0.1.0
    app.kubernetes.io/instance: test
    app.kubernetes.io/version: "1.16.0"
    app.kubernetes.io/managed-by: Helm
data:
  foo: "something_else"
```
This happens because in `helm upgrade`, the sub-chart is addressed using its `alias` and the values are taken from the `childzzz` (the sub-chart `alias`) key, as opposed from `helm install` where the values are taken from the `child` (the sub-chart `name`) key.

**Comment by bacongobbler** [comment]
slating this for 3.0.0 to evaluate whether or not this behaviour is in line with Helm 2, or if this is a regression that needs to be fixed prior to the final release.

**Comment by bacongobbler** [comment]
closed via #6084
